demo/gulpfile.js

```diff
@@ -62,18 +64,26 @@ gulp.task('serve-dist', ['build'], function() {
   runExpress(3000, DIST_DIR);
 });
 
-gulp.task('generate-service-worker-dev', function() {
-  var serviceWorkerFileContents = generateServiceWorkerFileContents(DEV_DIR, false);
-
-  return $.file('service-worker.js', serviceWorkerFileContents, {src: true})
-    .pipe(gulp.dest(DEV_DIR));
+gulp.task('generate-service-worker-dev', function(callback) {
+  generateServiceWorkerFileContents(DEV_DIR, false, function(error, serviceWorkerFileContents) {
+    if (error) {
+      return callback(error);
+    }
+    $.file('service-worker.js', serviceWorkerFileContents)
+      .pipe(gulp.dest(DEV_DIR));
```

Comment:
I've switched over to `fs.writeFile()` in https://github.com/jeffposnick/sw-precache/commit/00c2b0bc7ab785cae7446b3a166e83435c644f02